app/jitsi/models.py

```diff
@@ -39,37 +49,40 @@ def create(cls, username, avatar):
 
     @classmethod
     def leave_room(cls, user_id, room_name):
-        room = Room.query.filter_by(name=room_name).first()
-        user_location = UserLocation.query.filter_by(user_id=user_id, room_id=room.id).first()
-        if user_location:
-            db.session.delete(user_location)
-            db.session.commit()
+        user = User.query.filter_by(id=user_id).first()
+
+        if not user.room:
```

Comment:
don't we return `hallway` as room in some contexts? I seem to remember that. Could that cause issues here?

Reply:
`hallway` is only used in the return value `get_active_users_by_room()` for use in the map:
```
            if user.room:
                user_lists[user.room.name].append(user_dict)
            else:
                user_lists['hallway'].append(user_dict)
```
We never set the user's state as `hallway`.

Reply:
Okiedoke. Just worried that having two diff ways of representing the hallway state could lead to confusion down the road. But if it's fine now it may not be worth refactoring yet.